spec/models/youth_risk_assessment_spec.rb

```diff
@@ -3,7 +3,36 @@
 require 'rails_helper'
 
 RSpec.describe YouthRiskAssessment do
+  let(:from_location) { create(:location, :stc) }
+
   it { is_expected.to belong_to(:move) }
 
+  context 'with validations' do
+    it 'is valid if the move from location is from an stc' do
+      location = create(:location, :stc)
+      move = create(:move, from_location: location)
+      youth_risk_assessment = build(:youth_risk_assessment, move: move)
+
+      expect(youth_risk_assessment).to be_valid
+    end
+
+    it 'is valid if the move from location is from an sch' do
+      location = create(:location, :sch)
+      move = create(:move, from_location: location)
+      youth_risk_assessment = build(:youth_risk_assessment, move: move)
+
+      expect(youth_risk_assessment).to be_valid
+    end
+
+    it 'is invalid if the move from location is not from an sch or stc' do
+      location = create(:location, :prison)
+      move = create(:move, from_location: location)
+      youth_risk_assessment = build(:youth_risk_assessment, move: move)
+
+      expect(youth_risk_assessment).not_to be_valid
+      expect(youth_risk_assessment.errors.messages[:move]).to eq(["'from_location' must be from either a secure training centre or a secure children's home"])
+    end
```

Comment:
Could you add a test for `move` being nil please? There's some defensive coding against this in the `move_from_location` method but it doesn't look like it's all either necessary or being exercised.